File: sdks/java/src/main/java/dev/atap/sdk/exception/ATAPNotFoundException.java
package dev.atap.sdk.exception;

import dev.atap.sdk.model.ProblemDetail;

/**
 * Resource not found (404).
 */
public class ATAPNotFoundException extends ATAPException {

    private final ProblemDetail problem;

    public ATAPNotFoundException(String message, ProblemDetail problem) {
        super(message, 404);
        this.problem = problem;
    }

    public ATAPNotFoundException(String message) {
        this(message, null);
    }

    public ProblemDetail getProblem() {
        return problem;
    }
}

File: sdks/php/src/Exception/ATAPNotFoundException.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Exception;

use Atap\Sdk\Model\ProblemDetail;

/**
 * Resource not found (404).
 */
class ATAPNotFoundException extends ATAPException
{
    public function __construct(
        string $message = 'Not found',
        public readonly ?ProblemDetail $problem = null,
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, 404, $previous);
    }
}
